otj-kafka-streams/src/main/java/com/opentable/kafka/builders/OTKafkaClientSupplier.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.opentable.kafka.builders;
+
+import java.util.Map;
+import java.util.function.Supplier;
+
+import javax.inject.Inject;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.ByteArraySerializer;
+import org.apache.kafka.streams.KafkaClientSupplier;
+
+/**
+ * TODO: Food for thought
+ * For supporting Kafka Streams. A work in progress
+ * - See original DefaultKafkaClientSupplier
+ * - needs testing
+ * - naming is dubious, sigh
+ * - expose as spring bean
+ * - metrics namespace stability
+ * - Inability to edit properties EXCEPT via Map, which isn't consistent with others
+ * (The "fix" to this is expose a composite builder delegating to both consumer and producer builders, but
+ * that seems messy)
+ * KafkaStreams streams = new KafkaStreams(builder.build(), new StreamsConfig(config), supplier);
+ */
+public class OTKafkaClientSupplier implements KafkaClientSupplier {
+    private final KafkaBuilderFactoryBean kafkaBuilderFactoryBean;
+    private final Supplier<String> consumerNamer;
+    private final Supplier<String> producerNamer;
+
+
+    @Inject
+    public OTKafkaClientSupplier(KafkaBuilderFactoryBean kafkaBuilderFactoryBean, Supplier<String> consumerNamer, Supplier<String> producerNamer) {
+        this.kafkaBuilderFactoryBean = kafkaBuilderFactoryBean;
+        this.consumerNamer = consumerNamer;
+        this.producerNamer = producerNamer;
+    }
+
+    @Override
+    public AdminClient getAdminClient(final Map<String, Object> config) {
```

Comment:
Should access to the admin client really be on the root `KafkaClientSupplier` interface? `AdminClient` has a lot of power and I don't know if we should expose it here.

Reply:
Look at DefaultClienTSupplier

You’ll find this is copy pastr